### PR TITLE
[#610] Fix deleted object removal

### LINKED ISSUE #610: Inconsistent behavior of delete-object operation
## Steps to Reproduce (for bugs)
The issue is intermittent and does not occur on every run. I am observing it in about 50% of runs.

**Environment is:** dev-env with s3 gateway configured with multiple peers and tree service.
We are using the following environment variables in [docker-compose.yaml](https://github.com/nspcc-dev/neofs-dev-env/blob/master/services/s3_gate/docker-compose.yml) of s3 gateway:
```
    environment:
      - S3_GW_WALLET=/wallet.json
      - S3_GW_WALLET_PASSPHRASE=s3
      - S3_GW_TLS_KEY_FILE=/tls.key
      - S3_GW_TLS_CERT_FILE=/tls.crt
      - S3_GW_LISTEN_ADDRESS=s3.${LOCAL_DOMAIN}:8080
      - S3_GW_LISTEN_DOMAINS_0=s3.${LOCAL_DOMAIN}
      - S3_GW_PEERS_0_ADDRESS=s01.${LOCAL_DOMAIN}:8080
      - S3_GW_PEERS_0_WEIGHT=0.2
      - S3_GW_PEERS_1_ADDRESS=s02.${LOCAL_DOMAIN}:8080
      - S3_GW_PEERS_1_WEIGHT=0.2
      - S3_GW_PEERS_2_ADDRESS=s03.${LOCAL_DOMAIN}:8080
      - S3_GW_PEERS_2_WEIGHT=0.2
      - S3_GW_PEERS_3_ADDRESS=s04.${LOCAL_DOMAIN}:8080
      - S3_GW_PEERS_3_WEIGHT=0.2
      - S3_GW_TREE_SERVICE=s01.${LOCAL_DOMAIN}:8080. # <--- this is the only difference from standard config
```

The issue can be reproduced both with AWS CLI and with boto3 client. Steps are:
* Create new S3 bucket
* Enable versioning on the bucket
* Put object to the bucket
* Put a new version of the object to the bucket
* Delete object (providing `key`, but not `version` in the request)

Please, find attached logs for automated test run:
[aws-cli-log.txt](https://github.com/nspcc-dev/neofs-s3-gw/files/9161744/aws-cli-log.txt)
[boto3-log.txt](https://github.com/nspcc-dev/neofs-s3-gw/files/9161745/boto3-log.txt)

## Expected Behavior
The response from object-delete operation contains populated `VersionId` field.

## Current Behavior
Sometimes (about 50% of cases) response does not contain `VersionId`.

## Your Environment
Same as in https://github.com/nspcc-dev/neofs-s3-gw/issues/601:

* Version used:
```
COMMAND: 'neo-go --version'
OUTPUT:
 neo-go version 0.98.5

OMMAND: 'neofs-cli --version'
OUTPUT:
 Version: v0.28.1 
Build: 2022-05-05T09:07:45+00:00 
Debug: false

COMMAND: 'neofs-authmate --version'
OUTPUT:
 NeoFS gate authentication manager version v0.21.1

COMMAND: 'aws --version'
OUTPUT:
 2.2.0
AWS CLI v2 command: /root/.awscliv2/binaries/aws
aws-cli/2.7.11 Python/3.9.11 Linux/5.10.0-15-amd64 exe/x86_64.debian.11 prompt/off
```
* Server setup and configuration:
```
<project name="morph-crest.git" path="morph-crest" revision="a915cb945b7571c6098a21185bba620f7e0516eb" upstream="master" dest-branch="master" />
<project name="neo-go.git" path="neo-go" revision="b21a3c992de510c0598828fd9b6db9b35af33f3f" upstream="add-dbft-watchdog" dest-branch="add-dbft-watchdog" />
<project name="neo-preodolenie.git" path="neo-preodolenie" revision="a6fb56b26392cf837940483d09dcc6db272d9427" upstream="master" dest-branch="master" />
<project name="neofs-contract.git" path="neofs-contract" revision="a92ad8b8f0e7f5265bced249949ea7cbc9fdad75" upstream="refs/tags/v0.15.2" dest-branch="refs/tags/v0.15.2" />
<project name="neofs-dev-env.git" path="neofs-dev-env" revision="4f53093b186e8f4f5f8dd6d2946a9ca699235907" upstream="internal_exp_b" dest-branch="internal_exp_b" />
<project name="neofs-http-gw.git" path="neofs-http-gw" revision="6ce5fe80255ac2085031c57ddd88fbd1fd5b1219" upstream="refs/tags/v0.21.0" dest-branch="refs/tags/v0.21.0" />
<project name="neofs-keywords.git" path="neofs-testcases/neofs-keywords" revision="98b7f2601e5cc4f67184d4321da56df5531b68d5" upstream="internal_tmp_b" dest-branch="internal_tmp_b" />
<project name="neofs-keywords.git" path="neofs-keywords" revision="58650099362f05b51b97f04e2244a4ba20f88351" upstream="master" dest-branch="master" />
<project name="neofs-locode-db.git" path="neofs-locode-db" revision="ffd8516fccb63c9955f1d4202fb8f5b858188aa1" upstream="refs/tags/v0.2.1" dest-branch="refs/tags/v0.2.1" />
<project name="neofs-node.git" path="neofs-node" revision="34d20fd592f24d7b16fc4b119ad7675d6e01baf7" upstream="experimental" dest-branch="experimental" />
<project name="neofs-s3-gw.git" path="neofs-s3-gw" revision="03a68d0f9fc2ee020596c55445e86e493a5da5dd" upstream="tree-service" dest-branch="tree-service" />
<project name="neofs-testcases.git" path="neofs-testcases" revision="bed41402eedaed234349bfab2e22331c061a415b" upstream="internal_tmp_b" dest-branch="internal_tmp_b" />
```

### PR DESCRIPTION
**Pull request by KirillovDenis**
Fix tests:
* `s3tests_boto3.functional.test_s3.test_multi_object_delete`
* `s3tests_boto3.functional.test_s3.test_multi_objectv2_delete`

Also don't suppress error on object removal (except `not found` error)

close #610 
Signed-off-by: Denis Kirillov <denis@nspcc.ru>

### CODE REVIEW

File: api/handler/delete.go [hunk 1]
```diff
@@ -86,14 +86,10 @@ func (h *handler) DeleteObjectHandler(w http.ResponseWriter, r *http.Request) {
...
-			h.logAndSendError(w, "object is locked", reqInfo, errors.GetAPIError(errors.ErrAccessDenied))
-			return
+			deletedObject.Error = errors.GetAPIError(errors.ErrAccessDenied)
 		}
-		h.log.Error("could not delete object",
-			zap.String("request_id", reqInfo.RequestID),
-			zap.String("bucket_name", reqInfo.BucketName),
-			zap.String("object_name", reqInfo.ObjectName),
-			zap.Error(deletedObject.Error))
+		h.logAndSendError(w, "object is locked", reqInfo, deletedObject.Error)
```
Comment (masterSplinter01) [inline api/handler/delete.go:91]: Shouldn't we write `couldn't delete object` here?

**Comment by masterSplinter01** [comment]
I'm not sure but shouldn't we return an error if specified`version-id` is invalid ? I mean, Amazon returns an error if `version-id` is invalid:
```
$ aws s3api list-object-versions  --bucket testmarker  
{
  "Versions": [
    {
      "ETag": "\"bb285fe38300f07d7b755e3b45c0020d\"",
      "Size": 87422,
      "StorageClass": "STANDARD",
      "Key": "a",
      "VersionId": "BIhTJKZORje6nIPQ7jXFEIoGB.vgR6mm",
      "IsLatest": true,
      "LastModified": "2022-07-26T10:20:25+00:00",
      "Owner": {
        "DisplayName": "ops",
        "ID": "68e12342e5971d09ced74634fc7cc0fc5f01419d06b76353d42ce3108367183b"
      }
    }
  ]
}

$  aws s3api delete-object --key a --version-id BIhTJKZORje6nIPQ7jXFEIoGB  --bucket testmarker  

An error occurred (InvalidArgument) when calling the DeleteObject operation: Invalid version id specified
```

And the gw:

```
$ aws s3api list-object-versions  --bucket test3 --endpoint-url http://127.0.0.1:8084

{
  "Versions": [
    {
      "ETag": "03719ec2b4558dc89c82c47283b4ab9dd17e6e56d1858f254892707491443536",
      "Size": 87422,
      "Key": "a",
      "VersionId": "E9s7ce7oj5XmUKDPdpekCeK7MrE72orm1mF3ayUdfeEv",
      "IsLatest": true,
      "LastModified": "2022-07-26T10:13:05+00:00",
      "Owner": {
        "DisplayName": "NbUgTSFvPmsRxmGeWpuuGeJUoRoi6PErcM",
        "ID": "NbUgTSFvPmsRxmGeWpuuGeJUoRoi6PErcM"
      }
    }
  ]
}

$  aws s3api delete-object --key a --version-id E9s7ce7oj5XmUKDPdpekCeK7MrE72orm1m  --bucket test3  --endpoint-url http://127.0.0.1:8084
{
  "DeleteMarker": true,
  "VersionId": "E9s7ce7oj5XmUKDPdpekCeK7MrE72orm1m"
}
```

But we don't create a node with `version-id` from a request, so it's optional.



Also, it's confusing a little that we receive a response with `DeleteMarker: true` when delete-marker is not placed really.

**Comment by KirillovDenis** [comment]
> I mean, Amazon returns an error in that case

Yes and no. If you specify version-id like `BIhTJKZORje6nIPQ7jXFEIoGB.aaaaaa` Amazon returns no error

**Comment by KirillovDenis** [comment]
> The second, it's confusing a little that we receive a response with DeleteMarker: true when delete-marker is not placed really.

This is [aws behavior](https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObject.html#API_DeleteObject_ResponseSyntax) `x-amz-delete-marker:
Specifies whether the versioned object that was permanently deleted was (true) or was not (false) a delete marker.`:
```
$ aws s3api --profile aws delete-object --bucket dkirillov --key obj-dummy --version-id iNPx1Dla1wJ6ZXXc3Z43.ht5gWX3fOAV
{
    "DeleteMarker": true,
    "VersionId": "iNPx1Dla1wJ6ZXXc3Z43.ht5gWX3fOAV"
}
```

**Comment by KirillovDenis** [comment]
> When I try to delete a specific version I get another result

Sorry, confused with delete-marker removal

**Review by alexvanin — APPROVED** [review]

File: api/handler/delete_test.go [hunk 2]
```diff
@@ -1,10 +1,13 @@
 package handler
 
 import (
+	"bytes"
 	"net/http"
 	"net/url"
 	"testing"
 
+	"github.com/nspcc-dev/neofs-s3-gw/api/layer"
+
```
Comment (masterSplinter01) [inline api/handler/delete_test.go:10]: Empty line

**Review by masterSplinter01 — APPROVED** [review]

**Review by alexvanin — APPROVED** [review]

**Pull request merged**